### PR TITLE
feanil/re configure catalog providers

### PR DESCRIPTION
**Pull request by feanil**
- feat: Allow guest access to backstage.
- feat: Update the Github Plugin catalog Backend module.
- feat: Update how we pull entities from github.

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -75,20 +75,24 @@ scaffolder:
...
+        organization: 'openedx'
+        filters:
+          branch: 'main'
+      masterBranches:
+        organization: 'openedx'
+        filters:
+          branch: 'master'
+      extraCatalogs:
+        organization: 'openedx'
+        catalogPath: 'catalog-extra/**/catalog-info.yaml'
```
Comment (e0d) [inline app-config.yaml:93]: Does this mean the sub-catalogs live outside the code for the related component in a top-level catalog-extra folder?

File: packages/app/src/App.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const app = createApp({
   apis,
   components: {
     SignInPage: props => (
-      <SignInPage {...props} auto provider={githubProvider} />
+      <SignInPage {...props} auto providers={['guest', githubProvider]} />
```
Comment (e0d) [inline packages/app/src/App.tsx:52]: Is guest provider read only?

**Review by e0d — COMMENTED** [review]
Couple questions I'd like to understand.

File: app-config.yaml (hunk 1 above)
Comment (feanil) [reply app-config.yaml:93]: yea, that's right, this is where Dave is putting the LTI catalog file and we can add other non physical components we want to track here as well.

File: packages/app/src/App.tsx (hunk 2 above)
Comment (feanil) [reply packages/app/src/App.tsx:52]: Ah, good call, it is definitely not, It looks like adding a read-only capability might be more complicated than I initially thought.  It looks like the backstage demo app gets around this by removing the `register` view from the frontend.

Comment (feanil) [reply packages/app/src/App.tsx]: I removed this part from this PR so we can land the rest of it.

**Review by e0d — APPROVED** [review]
Looks good with the guest removed.

**Pull request merged**